coil-core/src/androidMain/kotlin/coil3/transform/transformations.kt

```diff
@@ -0,0 +1,37 @@
+package coil3.transform
+
+import android.graphics.Bitmap
+import android.graphics.BitmapShader
+import android.graphics.Matrix
+import android.graphics.Paint
+import android.graphics.Shader
+import coil3.decode.DecodeUtils
+import coil3.size.Scale
+
+/**
+ * Create a [Paint] that draws [input] centered and scaled to fit inside the output dimensions.
+ */
+internal fun newBitmapShaderPaint(
```

Comment:
Maybe fine since this function has params that hint to what it's doing, but just seeing the words `newBitmapShaderPaint` doesn't really outline what this does. Could change to something like `newScaledShaderPaint()` to help clarify.